matrix_openCL/Project1/main.cpp

```diff
@@ -0,0 +1,88 @@
+#define __CL_ENABLE_EXCEPTIONS
+#include <CL/cl.hpp>
+#include <utility>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <array>
+using namespace std;
+using namespace cl;
+
+int main() {
+    // Create the two input vectors
+    const int LIST_SIZE = 1000;
+    array<int, LIST_SIZE> A = {};
+    array<int, LIST_SIZE> B = {};
+    for (int i = 0; i < LIST_SIZE; i++) {
+        A[i] = i;
+        B[i] = LIST_SIZE - i;
+    }
+
+    try {
+        // Get available platforms
+        vector<Platform> platforms;
+        Platform::get(&platforms);
+
+        // Select the default platform and create a context using this platform and the GPU
+        cl_context_properties cps[3] = {
+            CL_CONTEXT_PLATFORM,
+            (cl_context_properties)(platforms[0])(),
+            0
+        };
+        Context context(CL_DEVICE_TYPE_GPU, cps);
+
+        // Get a list of devices on this platform
+        vector<Device> devices = context.getInfo<CL_CONTEXT_DEVICES>();
+
+        // Create a command queue and use the first device
+        CommandQueue queue = CommandQueue(context, devices[0]);
+
+        // Read source file
+        std::ifstream sourceFile("vector_adds_kernel.cl");
+        std::string sourceCode(
+            std::istreambuf_iterator<char>(sourceFile),
+            (std::istreambuf_iterator<char>()));
+        Program::Sources source(1, std::make_pair(sourceCode.c_str(), sourceCode.length() + 1));
+
+        // Make program of the source code in the context
+        Program program = Program(context, source);
+
+        // Build program for these specific devices
+        program.build(devices);
+
+        // Make kernel
+        Kernel kernel(program, "vector_add");
```

Comment:
if i change "vector_add" to anything else, it doesn't work. I don't understand why.

Reply:
because that's the function defined in vector_adds_kernel.cl